base/src/Data/Macaw/AbsDomain/AbsState.hs

```diff
@@ -93,8 +92,36 @@ import           Data.Macaw.Types
 ------------------------------------------------------------------------
 -- Utilities
 
-addOff :: NatRepr w -> Integer -> Integer -> Integer
-addOff w o v = toUnsigned w (o + v)
+-- | A helper for computing new signed offsets for values of type 'StackOffsetAbsVal'
+--
+-- This is a bit fiddly because 'StackOffsetAbsVal' has an 'Int64' as its second
+-- value, so we need to be very careful when converting from bitvectors (encoded
+-- as 'Integer') to 'Int64'.
+addSignedOffset
+  :: NatRepr w
+  -> Int64   -- ^ The current offset of a 'StackOffsetAbsVal'
+  -> Integer -- ^ The offset to add (to the current offset)
+  -> Int64
+addSignedOffset _w curOff i =
+  curOff + fromIntegral i
+
+-- | Add an offset represented as a bitvector to the current offset of a 'StackOffsetAbsVal'
+--
+-- Like 'addSignedOffset', this is tricky because the result is an 'Int64' while
+-- bitvectors are represented as raw 'Integer'.
+--
+-- This handles the interpretation of the bitvector as an 'Int64'. It uses the
+-- signed conversion function from the 'Data.Parameterized.NatRepr' module. This
+-- takes the low @w@ bits of the Integer and represents it in twos-complement so
+-- that the conversion to an 'Int64' properly sign extends.
+addSignedOffsetBV
+  :: (1 <= w)
+  => NatRepr w
+  -> Int64 -- ^ The current offset of a 'StackOffsetAbsVal'
+  -> Integer -- ^ A *bitvector* of width @w@ represented as an 'Integer' to add to the current offset; it will be interpreted as signed
```

Comment:
Would it not be more direct just to use a `BV w` here instead of an `Integer`? Half of `addSignedOffsetBV`'s call sites have access to a `BV` anyway, so it doesn't look like you're saving yourself that many keystrokes by using an `Integer` here.